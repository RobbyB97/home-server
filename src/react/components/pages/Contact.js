import React from 'react'
import {connect} from 'react-redux'



const Contact = ({dispatch, nav, title}) => (
	<div id="contact">
		Contact component
	</div>
)



const mapStateToProps = (state) => {
	return {
		nav: state.navigation.nav,
		title: state.title.title,
	}
}

export default connect(mapStateToProps)(Contact)