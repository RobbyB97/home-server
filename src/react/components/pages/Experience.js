import React from 'react'
import {connect} from 'react-redux'

import Job from '../content/Job'
import PrevPage from '../common/PrevPage'
import NextPage from '../common/NextPage'


export class Experience extends React.Component {
    
    render() {
        return (
            <div id="experience">
                
                Experience component
                <Job
                    title="Finalsite"
                    text="Job"
                    image="/dist/images/icons/finalsite.png">
                </Job>

                <PrevPage
                    text="Skills"
                    route="/skills/"
                    float={true}
                />

                <NextPage 
                    text="Projects"
                    route="/projects/"
                    float={true}
                />
            </div>  
        )
    }
}


const mapStateToProps = (state) => {
	return {
		nav: state.navigation.nav,
        title: state.title.title,
        splash: state.home.splash
	}
}

const mapDispatchToProps = (dispatch) => ({

})


export default connect(mapStateToProps, mapDispatchToProps)(Experience)