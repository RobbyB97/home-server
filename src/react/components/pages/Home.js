import React from 'react'
import {connect} from 'react-redux'

import Attribute from '../content/Attribute'


export class Home extends React.Component {

    render() {

        const fastAttribute = {
            image: "/dist/images/icons/speed_icon.png",
            title: "Fast",
            text: "Fully minified files and an optimized structure guarantee your website will load quickly, even on inconsistent internet connections."
        }

        const responsiveAttribute = {
            image: "/dist/images/icons/responsive_icon.png",
            title: "Responsive",
            text: "Mobile first development built up to 4k screens guarantees that your website will be elegant and functional on any device."
        }

        const beautifulAttribute = {
            image: "/dist/images/icons/paint_icon.png",
            title: "Beautiful",
            text: "A thorough, UX focused design process and open communication guarantees your website will look as beautiful as you envision it."
        }

        const accessibleAttribute = {
            image: "/dist/images/icons/accessibility_icon.png",
            title: "Accessible",
            text: "Full compliance with ADA standards guarantees that your website will be accessible and easy to use for everyone."
        }
        
        return (
            <div id="home">
                
                <div className={this.props.splash ? "home__background phone one": "home__background phone two"}></div>
                <div className={this.props.splash ? "home__background tablet one": "home__background tablet two"}></div>
                <div className={this.props.splash ? "home__background desktop one": "home__background desktop two"}></div>
                <div className={this.props.splash ? "home__background huge one": "home__background huge two"}></div>

                {!this.props.splash &&
                    <div className="home__attributes">
                        <Attribute {...fastAttribute} />
                        <Attribute {...responsiveAttribute} />
                        <Attribute {...beautifulAttribute} />
                        <Attribute {...accessibleAttribute} />
                    </div>
                }
                
            </div>
        )
    }
}


const mapStateToProps = (state) => {
    return {
      nav: state.navigation.nav,
      title: state.title.title,
      home: state.home.home,
      homePage: state.home.homePage,
      splash: state.home.splash
    }
}

const mapDispatchToProps = (dispatch) => ({

})


export default connect(mapStateToProps, mapDispatchToProps)(Home)