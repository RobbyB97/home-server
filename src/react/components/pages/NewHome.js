import React from 'react'


export class NewHome extends React.Component {
    render() {
        return (
            <div id="newHome">                
                <div className="newHome__content">
                    <div className="newHome__head">
                        <div className="newHome__name">
                            <h2>Robby Bergers</h2>
                        </div>
                        <div className="newHome__title">
                            <h3>Front End Developer</h3>
                        </div>
                    </div>

                    <div className="newHome__body">

                    </div>
                </div>

                <div className="newHome__backdrop"></div>
            </div>
        )
    }
}


export default NewHome