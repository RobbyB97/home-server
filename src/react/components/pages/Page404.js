import React from 'react'
import {NavLink} from 'react-router-dom'


export class Page404 extends React.Component {
    
    render() {
        return (
            <div id="page404">
                <div className="page404__backdrop"></div>
                <div className="page404__content">
                    <div className="page404__image">
                        <img src="/dist/images/wildcard.png" />
                    </div>

                    <div className="page404__text">
                        Uh oh, looks like this page doesn't exist.
                    </div>

                    <div className="page404__home-button">
                        <NavLink to="/">
                            <button>
                                Return home
                            </button>
                        </NavLink>
                    </div>
                </div>
            </div>
        )
    }
}


export default Page404