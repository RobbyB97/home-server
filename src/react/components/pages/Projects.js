import React from 'react'
import {connect} from 'react-redux'

import PrevPage from '../common/PrevPage'
import NextPage from '../common/NextPage'
import Project from '../content/Project'

import {
    project_ytIframe, 
    project_thisWebsite, 
    project_webDevBootcamp
} from '../../data/content/Project'


export class Projects extends React.Component {

    render() {
        return (
            <div id="projects">
                Projects component

                <Project {...project_ytIframe}/>
                <Project {...project_thisWebsite}/>
                <Project {...project_webDevBootcamp}/>
                
                <PrevPage
                    text="Experience"
                    route="/experience/"
                    float={true}
                />

                <NextPage 
                    text="Resume"
                    route="/resume/"
                    float={true}
                />
            </div>
        )
    }
}


const mapStateToProps = (state) => {
    return {
      nav: state.navigation.nav,
      title: state.title.title,
    }
}

const mapDispatchToProps = (dispatch) => {
    return {

    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Projects)