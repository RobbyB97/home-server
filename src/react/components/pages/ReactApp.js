import React from 'react'

import Header from '../Header'
import StyleGuide from './StyleGuide'
import Card from '../Card'

import {Link, NavLink} from 'react-router-dom'

export default class ReactApp extends React.Component {

    render() {
        return (
            <div>
                <section className="space home-cover"></section>
                <div className="home-text one">
                        <h2>Home-text 1</h2>
                    </div>
                    <div className="home-text two">
                        <h2>Home-text 2</h2>
                    </div>
                    <Card
                        link="https://www.github.com/robbyb97"
                        image="/assets/images/favicon.png"
                        title="My Github"
                        subtext="View my projects"
                    />
                <div className="home-cover">
                </div>
            </div>
        )
    }

}