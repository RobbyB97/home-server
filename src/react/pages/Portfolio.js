/* Libraries */
import React from 'react'
import Particles from 'react-particles-js'


/* Components */
import MobileNavigation from '../components/ui/MobileNavigation'
import DesktopNavigation from '../components/ui/DesktopNavigation'
import HomeButton from '../components/ui/HomeButton'


export class Portfolio extends React.Component {

    render() {
        return (
            <div id="#Portfolio">

                {/* UI */}
                <HomeButton internal />
                <DesktopNavigation />
                <MobileNavigation />

                {/* Landing section */}
                <div className="portfolio__landing">

                    <div className="portfolio__tagLine">
                        <h2 className="portfolio__tagLineTitle">
                            Robby Bergers
                        </h2>
                        <p className="portfolio__tagLineText one">
                            Designer
                        </p>
                        <p className="portfolio__tagLineText two">
                            Developer
                        </p>
                    </div>

                    <Particles 
                        width="100vw"
                        height="100vh"
                        className="portfolio__particles one"
                        params={{
                            particles: {
                                number: {
                                    value: 150,
                                    density: {
                                        enable: true,
                                        value_area: 700
                                    }
                                },
                                line_linked: {
                                    enable: true,
                                    distance: 60,
                                    color: '#808080',
                                    opacity: 0.5,
                                    width: 1
                                },
                                color: {
                                    value: '#000'
                                }
                            }
                        }}
                    />

                    <div className="portfolio__topShapes"></div>
                    <div className="portfolio__landingShape one"></div>
                    <div className="portfolio__landingShape two"></div>
                    <div className="portfolio__landingShape three"></div>
                    <div className="portfolio__landingShape four"></div>
                    <div className="portfolio__landingShape five"></div>
                    <div className="portfolio__landingBottom">
                        
                        <div>
                            Hire me on <a href="">Fiverr!</a>
                        </div>
                    </div>
                </div>

                {/* About section */}                
                <div id="about" className="portfolio__about">
                    
                    {/* Background fanciness top */}
                    <div className="portfolio__aboutShape one"></div>
                    <div className="portfolio__aboutShape two"></div>
                    <div className="portfolio__aboutShape three"></div>
                    <div className="portfolio__aboutShape four"></div>
                    <div className="portfolio__aboutShape five"></div>
                </div>

                {/* Services section */}
                <div id="services" className="portfolio__services">
                
                </div>

                <div id="skills" className="portfolio__skills">

                </div>
                <div id="experience" className="portfolio__experience">

                </div>
                <div id="projects" className="portfolio__projects">

                </div>
            </div>
        )
    }
}


export default Portfolio