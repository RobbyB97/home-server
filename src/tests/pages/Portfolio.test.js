import React from 'react'
import {shallow} from 'enzyme'

import {Portfolio} from '../../react/pages/Portfolio'


test('Render Portfolio', () => {
    const wrapper = shallow(<Portfolio />)
    expect(wrapper).toMatchSnapshot()
})