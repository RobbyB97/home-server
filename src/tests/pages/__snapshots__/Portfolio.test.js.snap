// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Portfolio 1`] = `
<div
  id="#Portfolio"
>
  <Connect(HomeButton)
    internal={true}
  />
  <DesktopNavigation />
  <Connect(MobileNavigation) />
  <div
    className="portfolio__landing"
  >
    <div
      className="portfolio__tagLine"
    >
      <h2
        className="portfolio__tagLineTitle"
      >
        Robby Bergers
      </h2>
      <p
        className="portfolio__tagLineText one"
      >
        Designer
      </p>
      <p
        className="portfolio__tagLineText two"
      >
        Developer
      </p>
    </div>
    <t
      className="portfolio__particles one"
      height="100vh"
      params={
        Object {
          "particles": Object {
            "color": Object {
              "value": "#000",
            },
            "line_linked": Object {
              "color": "#808080",
              "distance": 60,
              "enable": true,
              "opacity": 0.5,
              "width": 1,
            },
            "number": Object {
              "density": Object {
                "enable": true,
                "value_area": 700,
              },
              "value": 150,
            },
          },
        }
      }
      style={Object {}}
      width="100vw"
    />
    <div
      className="portfolio__topShapes"
    />
    <div
      className="portfolio__landingShape one"
    />
    <div
      className="portfolio__landingShape two"
    />
    <div
      className="portfolio__landingShape three"
    />
    <div
      className="portfolio__landingShape four"
    />
    <div
      className="portfolio__landingShape five"
    />
    <div
      className="portfolio__landingBottom"
    >
      <div>
        Hire me on 
        <a
          href=""
        >
          Fiverr!
        </a>
      </div>
    </div>
  </div>
  <div
    className="portfolio__about"
    id="about"
  >
    <div
      className="portfolio__aboutShape one"
    />
    <div
      className="portfolio__aboutShape two"
    />
    <div
      className="portfolio__aboutShape three"
    />
    <div
      className="portfolio__aboutShape four"
    />
    <div
      className="portfolio__aboutShape five"
    />
  </div>
  <div
    className="portfolio__services"
    id="services"
  />
  <div
    className="portfolio__skills"
    id="skills"
  />
  <div
    className="portfolio__experience"
    id="experience"
  />
  <div
    className="portfolio__projects"
    id="projects"
  />
</div>
`;
